src/test/test_sign.py

```diff
@@ -377,5 +382,49 @@ def add_by_final_vbf(a_hex, b_hex):
         self.assertEqual(mul(zero, scalar), (WALLY_OK, zero_hex))
         self.assertEqual(mul(zero, negative), (WALLY_OK, zero_hex))
 
+    def test_bip340_sigs(self):
+
+        made_sig = False
+
+        for case in self.get_bip340_sign_cases():
+            priv_key = make_cbuffer(case["priv_key"])
+            pub_key = make_cbuffer(case["pub_key"])
+            aux_rand = make_cbuffer(case["aux_rand"])
+            msg = make_cbuffer(case["msg"])
+            sig = make_cbuffer(case["sig"])
+            verification = case["verification"]
+
+            sig_out = self.cbufferize(['00' * 64])[0]
+
+            # We don't expose aux_rand, so only test default value for now
+            if priv_key[1] == 32 and aux_rand[0] == bytes(32):
+                flags = FLAG_SCHNORR
+                self.assertEqual(WALLY_OK, self.sign(priv_key[0], msg[0], flags, sig_out))
+                self.assertEqual(sig[0], sig_out)
+                made_sig = True
+
+            if priv_key[1] == 32:
+                ret = wally_ec_sig_from_bytes_len(priv_key[0], priv_key[1],
+                                                  msg[0], msg[1], FLAG_SCHNORR)
+                self.assertEqual(ret, (WALLY_OK, EC_SIGNATURE_LEN))
+
+                # Wrong size privkey
+                ret = wally_ec_sig_from_bytes_len(priv_key[0][:-1], priv_key[1] - 1,
```

Comment:
you should remove the `[:-1]` from the first argument here, only the pointer is being passed so there's no need to take a slice.

Reply:
done